docs/introduction.rst

```diff
@@ -107,17 +109,49 @@ structure could be like a simplified XML DOM-tree with following specification:
   (e.g. the unit of the data in the node, etc.). To make things simple, the
   attributes should be text attributes, like in XML.
 
-The sender would assemble a tree with the necessary information and transmit it via
-the transport layer. The receiver would then query the received tree, look
+The sender would assemble a tree with the necessary information and transmit it
+via the transport layer. The receiver would then query the received tree, look
 for the presence / absence of given nodes and extract the necessary information
 from the tree.
 
 I have already started a small C-library with this functionality, the `saydx
-library <https://github.com/saydx/saydx>`_. Although not finished yet, it could
-be used as the message layer. It would provide the basic infrastructure for tree
-manipulation, as well as routines to read and write trees to file or to pass
-them from C to Fortran and vice-versa. Combined with cslib, it could cover the
-functionality of the first two layers.
+[səeɪdɪks] library <https://github.com/saydx/libsaydx>`_. Although not finished
+yet, it could be used as the message layer. It would provide the basic
+infrastructure for tree manipulation, as well as routines to read and write
+trees to file or to pass them from C to Fortran and vice-versa. Combined with
+cslib, it could cover the functionality of the first two layers.
+
+
+Array types
+^^^^^^^^^^^
+
+The message layer should understand / support following array data types:
+
+* real numbers (4, 8 and eventually 16 bytes)
+* complex numbers (composed of two real numbers)
```

Comment:
Is the plan to support C99 and std::complex for C and C++ bindings respectively, or return real and complex  as separate arrrays?

Reply:
We should try to support C99's complex type. If not possible for whatever reasons, we would return an array, where each element just consists of two consecutive reals and the client must deal with that on the C-side. Returning real and imaginary parts as separate arrays would be sub-optimal, as those arrays are usually not contiguous in memory (e.g. when using C99 complex or Fortran complex representation).